google/cloud/aiplatform/metadata/metadata.py

```diff
@@ -0,0 +1,133 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+from typing import Dict
+
+from google.cloud.aiplatform.metadata.metadata_store import _MetadataStore
+from google.cloud.aiplatform.metadata.context import _Context
+from google.cloud.aiplatform.metadata.execution import _Execution
+from google.cloud.aiplatform.metadata.artifact import _Artifact
+
+
+class _MetadataService:
+    """Contains the exposed APIs to interact with the Managed Metadata Service."""
+
+    def __init__(self):
+        self._experiment = None
+        self._run = None
+        self._metrics = None
+
+    def set_experiment(self, experiment: str):
+        if not experiment:
+            raise ValueError(f"Invalid experiment {experiment}.")
+
+        store = _MetadataStore.get()
+        if not store:
+            logging.info(
+                f"Creating a default MetadataStore for experiment {experiment}"
+            )
+            _MetadataStore.create()
+
+        context = _Context.get(context_name=experiment)
+        if not context:
+            logging.info(f"Creating a Context for experiment {experiment}")
+            context = _Context.create(
+                context_id=experiment,
+                schema_title="system.Experiment",
+                schema_version="0.0.1",
+            )
+        self._experiment = context.name
+
+    def set_run(self, run: str):
```

Comment:
As I am thinking we do no need to expose a separate set_run methods to end user, I think it is probably more straightforward to combine all the experiment/run/metric artifact initialization steps into one init function so that we can check and create all resources, and also connect them together.

Reply:
The init currently calls set_experiment and set_run. So is basically creating everything, but it's still missing the piece of connecting them together. We can try to merge this asap, and then you can keep on adding the missing bits.

Reply:
Discussed offline. We can get this in, and then un-expose if not needed